src/components/burger-menu/BurgerMenu.jsx

```diff
@@ -1,6 +1,61 @@
-import React, { useState, useContext } from 'react';
+import React, { useContext, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { GlobalContext } from '../GlobalContext';
+import { AppRoutes, HeaderItems } from '../../config';
+import styles from './BurgerMenu.module.scss';
+import burgerMenuIcon from '../../menu-icon/menu-icon.svg';
+import closeMenuIcon from '../../menu-icon/close-menu-icon.svg';
+import cn from '../../createClassNames';
 
 export const BurgerMenu = () => {
-  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
+  const { amounts: { accumulations }, balanceState } = useContext(GlobalContext);
 
-}
\ No newline at end of file
+  const navigate = useNavigate();
+  const location = useLocation();
+  const pathName = location.pathname === '/' ? AppRoutes.MY_WALLET : location.pathname;
```

Comment:
instead of doing this check just add default route in your app
      `<Route path="*" element={<Navigate replace to="/my-wallet" />} />`
logic should really simple if user opens your app in root route>>>  /   you rederecting his to my wallet page